perun-core/src/main/java/cz/metacentrum/perun/core/impl/ExtSourceCSV.java

```diff
@@ -113,71 +122,111 @@ public static PerunBlImpl setPerunBlImpl(PerunBlImpl perun) {
 
         } catch (IOException ex) {
             log.error("IOException in getSubjectByLogin() method while parsing csv file", ex);
+            throw new InternalErrorException(ex);
         }
-
-        return null;
     }
 
+    /**
+     * Get subjects of group. These subjects will be stored in maps obtained in parameters.
+     *
+     *
+     * @param sess
+     * @param group Map of attributes used for quering the external source
+     * @param status Status about synchronization (lightweight, full). Not used in CSV external source.
+     * @param subjects list of maps, which contains attr_name-&gt;attr_value, e.g. firstName-&gt;Michal
+     * @return String containing status about data received from external source (e.g. data are sorted,
+     * @throws InternalErrorException
+     * @throws ExtSourceUnsupportedOperationException
+     */
     @Override
-    public List<Map<String, String>> getGroupSubjects(Map<String, String> attributes) throws InternalErrorException, ExtSourceUnsupportedOperationException {
+    public String getGroupSubjects(PerunSession sess, Group group, String status, List<Map<String, String>> subjects) throws InternalErrorException, ExtSourceUnsupportedOperationException {
```

Comment:
This is not correct. Method says "Get subjects of group" in javadoc. But instead of list of subjects, it returns some sort of internal status.